diff --git a/src/common.sh b/src/common.sh
index f3cfc07..047283b 100755
--- a/src/common.sh
+++ b/src/common.sh
@@ -207,9 +207,11 @@ function mount_image() {
   fi
   sudo mkdir -p $mount_path/dev/pts
   sudo mkdir -p $mount_path/proc
+  sudo mkdir -p $mount_path/sys
   sudo mount -o bind /dev $mount_path/dev
   sudo mount -o bind /dev/pts $mount_path/dev/pts
-  sudo mount -o bind /proc $mount_path/proc
+  sudo mount -o bind,ro /proc $mount_path/proc
+  sudo mount -o bind,ro /sys $mount_path/sys
 }
 
 function unmount_image() {
